Issue with Node builtins in worker: ModuleNotFoundError: Module not found: Error: Can't resolve 'child_process'

I'm using threads-plugin with Vue CLI and Electron, and forcing threads.js to use Web Workers instead of Node Workers as explained in https://github.com/andywer/threads.js/issues/199. My worker code uses some Node built-in modules, like child_process. When webpack tries to compile my code with a `require('child_process')` in the worker code, I get the error: `ModuleNotFoundError: Module not found: Error: Can't resolve 'child_process'`.

I know it should be possible to use node builtins in worker code in my setup, because:
1. Electron supports this with the `nodeIntegrationWorker`option: https://electronjs.org/docs/tutorial/multithreading
2. It works fine with https://github.com/nolanlawson/promise-worker and loading the worker using https://github.com/webpack-contrib/worker-loader. Thus, I suspect that something about the way the worker code is loaded with threads-plugin is causing the issue. (I am migrating from promise-worker to get some of the more advanced features of threads.js, like pools and Transfer)

Seems related to https://github.com/andywer/threads-plugin/issues/10, but I'm not using TypeScript so I made a separate issue. Sorry for being a bit light on code examples in this issue, but I'm not sure what would be most relevant to provide. Would it be helpful for me to provide my webpack config? (it's a bit long because it's mostly the Vue CLI config)

I have tried various modifications to the webpack config, including setting `externals: { child_process: 'child_process' }` and `node: false` but nothing has helped. Happy to try any suggestions, thanks! :)

@andywer Do you mean this fix https://github.com/andywer/threads.js/issues/194#issuecomment-576155972? You already suggested it to me in this [issue](https://github.com/andywer/threads.js/issues/199#issue-548158290) and it helped, but I'm experiencing a different one here.
Being more exact I have this error stack:
```
Module build failed (from ./node_modules/threads-plugin/dist/loader.js):
ModuleNotFoundError: Module not found: Error: Can't resolve 'child_process' in '/home/vlad/Work/desktop-generator-v3/node_modules/detect-libc/lib'
    at /home/vlad/Work/desktop-generator-v3/node_modules/webpack/lib/Compilation.js:925:10
    at /home/vlad/Work/desktop-generator-v3/node_modules/webpack/lib/NormalModuleFactory.js:401:22
    at /home/vlad/Work/desktop-generator-v3/node_modules/webpack/lib/NormalModuleFactory.js:130:21
    at /home/vlad/Work/desktop-generator-v3/node_modules/webpack/lib/NormalModuleFactory.js:224:22
    at /home/vlad/Work/desktop-generator-v3/node_modules/neo-async/async.js:2830:7
    at /home/vlad/Work/desktop-generator-v3/node_modules/neo-async/async.js:6877:13
    at /home/vlad/Work/desktop-generator-v3/node_modules/webpack/lib/NormalModuleFactory.js:214:25
    at /home/vlad/Work/desktop-generator-v3/node_modules/enhanced-resolve/lib/Resolver.js:213:14
    at /home/vlad/Work/desktop-generator-v3/node_modules/enhanced-resolve/lib/Resolver.js:285:5
    at eval (eval at create (/home/vlad/Work/desktop-generator-v3/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:13:1)
    at /home/vlad/Work/desktop-generator-v3/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:44:7
    at /home/vlad/Work/desktop-generator-v3/node_modules/enhanced-resolve/lib/Resolver.js:285:5
    at eval (eval at create (/home/vlad/Work/desktop-generator-v3/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:13:1)
    at /home/vlad/Work/desktop-generator-v3/node_modules/enhanced-resolve/lib/Resolver.js:285:5
    at eval (eval at create (/home/vlad/Work/desktop-generator-v3/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:25:1)
    at /home/vlad/Work/desktop-generator-v3/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:67:43
```
If I add `"browser": { "child_process": false },` to `detect-libc/package.json`, it changes to
```
Module build failed (from ./node_modules/threads-plugin/dist/loader.js):
ModuleNotFoundError: Module not found: Error: Can't resolve 'child_process' in '/home/vlad/Work/desktop-generator-v3/node_modules/sharp/lib'
```
And if I add the same to `sharp/package.json`, the error becomes
```
Module build failed (from ./node_modules/threads-plugin/dist/loader.js):
ModuleNotFoundError: Module not found: Error: Can't resolve 'constants' in '/home/vlad/Work/desktop-generator-v3/node_modules/fs-constants'
```

Not even sure if my issue is related to the one I'm posting to, but the first error seems the same as in the first post here

--- a/README.md
+++ b/README.md
@@ -100,6 +100,11 @@ When using TypeScript, make sure this setting is part of your TypeScript configu
 }
 ```
 
+## Electron
+
+To use standard Node packages in Electron web workers, make sure Node integration in workers is [turned on](https://www.electronjs.org/docs/tutorial/multithreading)
+and the plugin option [target](#target) is set to `electron-node-worker`.
+
 ## Options
 
 In most cases, no options are necessary to use WorkerPlugin.
@@ -151,6 +156,23 @@ module.exports = {
   ]
   <...>
 }
+```
+
+### `target` 
+
+Due to the way webpack works, it may be necessary to change the target environment of the workers.
+
+Possible values:
+
+- `electron-node-worker` use this if you are using Electron and compiling web workers that have node integration enabled. See [Electron Multithreading](https://www.electronjs.org/docs/tutorial/multithreading).
+
+Default value: `undefined`
+
+Example with electron node workers:
+```js
+new ThreadsPlugin({
+  target: 'electron-node-worker'
+})
 ```
 
 ## License

--- a/src/loader.js
+++ b/src/loader.js
@@ -18,6 +18,7 @@ import loaderUtils from 'loader-utils';
 import SingleEntryPlugin from 'webpack/lib/SingleEntryPlugin';
 import WebWorkerTemplatePlugin from 'webpack/lib/webworker/WebWorkerTemplatePlugin';
 import FetchCompileWasmTemplatePlugin from 'webpack/lib/web/FetchCompileWasmTemplatePlugin';
+import NodeTargetPlugin from 'webpack/lib/node/NodeTargetPlugin';
 import WORKER_PLUGIN_SYMBOL from './symbol';
 
 const NAME = 'WorkerPluginLoader';
@@ -57,6 +58,9 @@ export function pitch (request) {
 
   const workerCompiler = this._compilation.createChildCompiler(NAME, workerOptions, plugins);
   workerCompiler.context = this._compiler.context;
+  if (pluginOptions.target === "electron-node-worker") {
+    new NodeTargetPlugin().apply(workerCompiler);
+  }
   (new WebWorkerTemplatePlugin(workerOptions)).apply(workerCompiler);
   (new FetchCompileWasmTemplatePlugin({
     mangleImports: compilerOptions.optimization.mangleWasmImports

